### PR TITLE
adapt tls flag usage for upcoming release

### PR DESCRIPTION
**Pull request by frzifus**
Signed-off-by: Benedikt Bongartz <bongartz@klimlive.de>

## Which problem is this PR solving?
- From the upcoming version 1.34 the tls enabled flag is required when using tls options. https://github.com/jaegertracing/jaeger/pull/3030 

## Short description of the changes
- to avoid breaking existing setups, tls flags are extended by the missing enable flag if necessary.

---
cc @rubenvp8510 @pavolloffay

### CODE REVIEW

**Review by iblancasa — DISMISSED** [review]

File: apis/v1/jaeger_webhook.go [hunk 1]
```diff
@@ -110,3 +123,28 @@ func ShouldInjectOpenShiftElasticsearchConfiguration(s JaegerStorageSpec) bool {
 	_, ok := s.Options.Map()["es.server-urls"]
 	return !ok
 }
+
+var (
+	tlsFlag          = regexp.MustCompile("--.*tls.*")
+	tlsFlagIdx       = regexp.MustCompile("--.*tls")
+	tlsEnabledExists = regexp.MustCompile("--.*tls.enabled.*")
```
[the remark this answers is not in the document]
Comment (frzifus) [reply apis/v1/jaeger_webhook.go]: good point, its not needed.

File: apis/v1/jaeger_webhook.go [hunk 2]
```diff
@@ -110,3 +123,28 @@ func ShouldInjectOpenShiftElasticsearchConfiguration(s JaegerStorageSpec) bool {
...
+)
+
+func tlsMissingFlag(args []string) map[string]interface{} {
```
[the remark this answers is not in the document]
Comment (frzifus) [reply apis/v1/jaeger_webhook.go:137]: Originally, the function was meant to be. Then I noticed that outside the function the flag prefix is not available without parsing it again. (have extended the function with a comment).

**Review by rubenvp8510 — DISMISSED** [review]
LGTM

**Comment by frzifus** [comment]
lets wait, please

**Review by rubenvp8510 — DISMISSED** [review]
LGTM

**Comment by frzifus** [comment]
for some reason the default webhook changes are not taken into account.
```
1.6528167631550221e+09	INFO	jaeger-resource	default	{"name": "tls-streaming"}
1.6528167631554368e+09	DEBUG	controller-runtime.webhook.webhooks	wrote response	{"webhook": "/mutate-jaegertracing-io-v1-jaeger", "code": 200, "reason": "", "UID": "cf50fa92-ece6-4908-80d6-9f74b8bf093d", "allowed": true}
1.6528167631606963e+09	DEBUG	controller-runtime.webhook.webhooks	received request	{"webhook": "/validate-jaegertracing-io-v1-jaeger", "UID": "fd459665-cf11-4845-b88e-ecde34dc7394", "kind": "jaegertracing.io/v1, Kind=Jaeger", "resource": {"group":"jaegertracing.io","version":"v1","resource":"jaegers"}}
1.6528167631608965e+09	INFO	jaeger-resource	validate create	{"name": "tls-streaming"}
1.6528167631609068e+09	INFO	jaeger-resource	validate update	{"name": "tls-streaming"}
1.6528167631609724e+09	DEBUG	controller-runtime.webhook.webhooks	wrote response	{"webhook": "/validate-jaegertracing-io-v1-jaeger", "code": 403, "reason": "tls flags incomplete, got: [--kafka.producer.tls.key=/var/run/secrets/kafkauser/user.key --kafka.producer.topic=jaeger-spans --kafka.producer.authentication=tls --kafka.producer.brokers=my-cluster-kafka-bootstrap:9093 --kafka.producer.tls.ca=/var/run/secrets/cluster-ca/ca.crt --kafka.producer.tls.cert=/var/run/secrets/kafkauser/user.crt]", "UID": "fd459665-cf11-4845-b88e-ecde34dc7394", "allowed": false}
```

File: apis/v1/jaeger_webhook.go [hunk 3]
```diff
@@ -110,3 +140,28 @@ func ShouldInjectOpenShiftElasticsearchConfiguration(s JaegerStorageSpec) bool {
 	_, ok := s.Options.Map()["es.server-urls"]
 	return !ok
 }
+
+var (
+	tlsFlag          = regexp.MustCompile("--.*tls.*=")
```
Comment (frzifus) [inline apis/v1/jaeger_webhook.go:145]: added `=`

File: apis/v1/options.go [hunk 4]
```diff
@@ -88,6 +88,7 @@ func (o Options) MarshalJSON() ([]byte, error) {
 }
 
 func (o *Options) parse(entries map[string]interface{}) error {
+	o.json = nil
```
Comment (frzifus) [inline apis/v1/options.go:91]: added this to clear the json buffer

**Review by rubenvp8510 — APPROVED** [review]

**Pull request merged**